app/controllers/gifts_controller.rb

```diff
@@ -44,12 +44,16 @@ def toggle_rejected
     @gift = Gift.find(params[:id])
     # have to have the gift, be logged in, and the owner of the gift
     if @gift && current_user && @gift.user == current_user
-      @gift.toggle!(:rejected)
+      @gift.rejected = !@gift.rejected?
+      @gift.save!
```

Comment:
Whatever was causing it to fail before, it was failing silently. Maybe the validation issue wasn't really the problem if toggle skips validation, but "save!" throws an error if it fails, which means we'll at least be able to see what the problem is.